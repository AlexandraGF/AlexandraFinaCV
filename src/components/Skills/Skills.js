import React, { Component } from 'react';
import './Skills.scss';

class Skills extends Component {
    constructor(props) {
        super(props);
        this.state = {clicked: false};
    
        this.handleClick = this.handleClick.bind(this);
      }

      handleClick() {
        this.setState(state => ({
          clicked: !state.clicked
        }))
    };

    render() {
        const codingSkills = [ 
            {id: 1, name: 'ReactJs'},  
            {id: 2, name: 'Nodejs'}, 
            {id: 3, name: 'Javascript'}, 
            {id: 4, name: 'Ruby'}, 
            {id: 5, name: 'Jasmine'}, 
            {id: 6, name: 'RSpec'}, 
            {id: 7, name: 'Jest testing framework'}, 
            {id: 8, name: 'Git/GitHub'}, 
            {id: 9, name: 'TDD'}, 
            {id: 10, name: 'Pair Programming'}, 
            {id: 11, name: 'HTML'}, 
            {id: 12, name: 'CSS'},
            {id: 13, name: 'Heroku'},
        ]; 
        const spokenLanguages = [
            {id: 1, name: 'English', level: 'Excellent working proficiency'}, 
            {id: 2, name: 'Romanian', level: 'Native speaker'}, 
            {id: 3, name: 'Spanish', level: 'Conversational proficiency'},
        ];
        return (
        <div> 
            <div className="skillsHeading" onClick={this.handleClick}>
                <div className="skillsTitle">Skills</div>
                <div className="skillsIconImage">{!this.state.clicked ? "+" : "-"}</div>
            </div>
            <div>{this.state.clicked ? (
                <div>
                    <div className="skillsBody">
                        {codingSkills.map(function(item, i){
                            return <div className="listItemSkills" key={i}>* {item.name}</div>
                        })}
                    </div> 
                    <div className="languageBody">Spoken Languages
                        {spokenLanguages.map(function(item, i){
                    return <div className="listItemLanguage" key={i}>* {item.name}: {item.level}</div>
                    })}
                    </div>
                </div>) : "" }
            </div>
        </div>
        );
    }
}

export default Skills;
