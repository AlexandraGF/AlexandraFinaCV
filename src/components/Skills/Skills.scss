@import '../../variables.scss';

.skillsHeading {
    font-family: 'Allura', cursive;
    font-size: 30px;
    margin: 7% 7% 0 11%;
    padding-left: 6%;
    box-shadow: $boxShadow;
    border-radius: $borderRadius;
    display: inline-block;
    width: 80%;
}

.skillsTitle {
    width: 80%;
}

.skillsIconImage {
    width: 8%;
}

.skillsTitle, .skillsIconImage {
    flex-wrap: wrap;
    display: inline-flex;
}

.skillsHeading:hover {
    transform: $hoverSize;
    cursor: $cursorPointer;
}

.skillsBody {
    box-shadow: 5px 10px 12px #a2c3c3;
    display: inline-flex;
    flex-wrap: wrap;
    margin: 7% 4% 5% 12%;
    font-family: 'K2D', sans-serif;
}

.listItemSkills {
    margin: 0 5% 5% 5%;
    padding: 1% 1% 1%;
    box-shadow: $boxShadow;
}

.listItemLanguage {
    margin: 0 5% 5% 12%;
    padding: 1% 1% 1%;
    box-shadow: $boxShadow;
}
