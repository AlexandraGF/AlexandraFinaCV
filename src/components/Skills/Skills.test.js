import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

import Skills from './Skills';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Skills />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders text', () => {
    const wrapper = mount(<Skills />);
    expect(wrapper.text()).toContain("Skills");
    });
    
    it('renders Skills Component', () => {
    const tree = renderer.create(
    <Skills />
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
    })