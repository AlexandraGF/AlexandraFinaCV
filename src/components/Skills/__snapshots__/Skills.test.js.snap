// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App Component 1`] = `
<div>
  <div
    className="skillsHeading"
    onClick={[Function]}
  >
    <div
      className="skillsTitle"
    >
      Skills
    </div>
    <div
      className="skillsIconImage"
    >
      <img
        src="plus.png"
      />
    </div>
  </div>
  <div>
    
  </div>
</div>
`;

exports[`renders Skills Component 1`] = `
<div>
  <div
    className="skillsHeading"
    onClick={[Function]}
  >
    <div
      className="skillsTitle"
    >
      Skills
    </div>
    <div
      className="skillsIconImage"
    >
      <img
        src="plus.png"
      />
    </div>
  </div>
  <div>
    
  </div>
</div>
`;
